File: aura_admin/app/core/network/api-endpoints.ts
export const API_ENDPOINTS = {
    AUTH: {
        LOGIN: "/api/admin/login",
        CURRENT: "/api/admin/current",
        REGISTER: "/api/admin/register",
    },
    ACTIVITY_CATEGORIES: {
        BASE: "/api/admin/activity-categories",
        ACTIVE: "/api/admin/activity-categories/active",
        BY_ID: (id: string) => `/api/admin/activity-categories/${id}`,
        TOGGLE: (id: string) => `/api/admin/activity-categories/${id}/toggle-status`,
    },
    ACTIVITY_TYPES: {
        BASE: "/api/admin/activity-types",
        ACTIVE: "/api/admin/activity-types/active",
        BY_ID: (id: string) => `/api/admin/activity-types/${id}`,
        BY_CATEGORY: (categoryId: string) => `/api/admin/activity-types/category/${categoryId}`,
        GYM: "/api/admin/activity-types/gym",
        TOGGLE: (id: string) => `/api/admin/activity-types/${id}/toggle-status`,
    },
    GYM_EXERCISES: {
        BASE: "/api/admin/gym-exercises",
        ACTIVE: "/api/admin/gym-exercises/active",
        BY_ID: (id: string) => `/api/admin/gym-exercises/${id}`,
        BY_MUSCLE: (muscle: string) => `/api/admin/gym-exercises/muscle/${muscle}`,
        TOGGLE: (id: string) => `/api/admin/gym-exercises/${id}/toggle-status`,
    },
    SOS: {
        EVENTS: "/api/admin/sos/events",
        EVENT_BY_ID: (id: string) => `/api/admin/sos/events/${id}`,
        ACKNOWLEDGE: (id: string) => `/api/admin/sos/events/${id}/acknowledge`,
        RESOLVE: (id: string) => `/api/admin/sos/events/${id}/resolve`,
        STATS: "/api/admin/sos/stats",
        LIVE_SESSIONS: "/api/user/sos/live/active",
        LIVE_SESSION_BY_ID: (id: string) => `/api/user/sos/live/${id}`,
    },
    WELLNESS: {
        ALL: "/api/admin/wellness/all",
        STATS: "/api/admin/wellness/stats",
        DELETE: (id: string) => `/api/admin/wellness/${id}`,
        HIDE: (id: string) => `/api/admin/wellness/${id}/hide`,
        WARN: (id: string) => `/api/admin/wellness/${id}/warn`,
    },
    USERS: {
        BASE: "/api/admin/users",
        BY_ID: (id: string) => `/api/admin/users/${id}`,
    },
};

File: aura_app/lib/core/network/http/api_endpoints.dart
class ApiEndpoints {
  static const me = "/api/auth/me";
  static const login = "/api/auth/login";
  static const register = "/api/auth/register";
  static const userProfile = "/api/user";
  static const usernameAvailable = "/api/user/username-available";
  static const updateProfile = "/api/user/profile";
  static const uploadProfileImage = "/api/upload/profile-image";
  static const verifyEmailSend = "/api/user/verify-email/send";
  static const verifyEmailConfirm = "/api/user/verify-email/confirm";

  static const changePassword = "/api/user/password/change";
  static const forgotPassword = "/api/auth/password/forgot";
  static const resetPassword = "/api/auth/password/reset";

  static const dailyActivities = "/api/daily-activities";
  static const syncDailyActivities = "/api/daily-activities/sync";

  static const activityCategories = "/api/admin/activity-categories";
  static const activityCategoriesActive =
      "/api/admin/activity-categories/active";

  static const activityTypes = "/api/admin/activity-types";
  static const activityTypesActive = "/api/admin/activity-types/active";
  static String activityTypesByCategory(String categoryId) =>
      "/api/admin/activity-types/category/$categoryId";

  static const userActivities = "/api/user/activities";
  static String userActivitiesForUser(String userId) =>
      "/api/user/activities/$userId";
  static String userActivitiesForDate(String userId, String date) =>
      "/api/user/activities/$userId/date/$date";

  static const activityLogs = "/api/user/activity-logs";
  static String activityLogsForDate(String userId, String date) =>
      "/api/user/activity-logs/$userId/date/$date";

  static const reminders = "/api/user/reminders";
  static String remindersForUser(String userId) =>
      "/api/user/reminders/$userId";

  static const sosSettings = "/api/user/sos";
  static const sosMessage = "/api/user/sos/message";
  static const sosContacts = "/api/user/sos/contacts";
  static String sosContactById(String contactId) =>
      "/api/user/sos/contacts/$contactId";
  static const sosTrigger = "/api/user/sos/trigger";
  static const sosEvents = "/api/user/sos/events";

  static const wellnessFeed = "/api/user/wellness/feed";
  static const wellnessMyUpdates = "/api/user/wellness/my-updates";
  static const wellnessTrending = "/api/user/wellness/trending";
  static const wellnessUpdates = "/api/user/wellness/updates";
}
